include/seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp

```diff
@@ -0,0 +1,474 @@
+            std::ranges::iterator_t<view_type> host_iter{};
+        }; // class iterator_type
+
+    public:
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        constexpr alignment_column_type() = default; //!< Defaulted.
+        constexpr alignment_column_type(alignment_column_type const &) = default; //!< Defaulted.
+        constexpr alignment_column_type(alignment_column_type &&) = default; //!< Defaulted.
+        constexpr alignment_column_type & operator=(alignment_column_type const &) = default; //!< Defaulted.
+        constexpr alignment_column_type & operator=(alignment_column_type &&) = default; //!< Defaulted.
+        ~alignment_column_type() = default; //!< Defaulted.
+
+        /*!\brief Constructs from the derived type.
+         * \param[in] me  An instance of the derived type.
+         * \param[in] ref The underlying view referencing the current column in the matrix.
+         *
+         * \details
+         *
+         * This constructor is called by the derived type when invoking the function `initialise_column`.
+         */
+        constexpr alignment_column_type(derived_t & me, view_type ref) :
+            ref{std::move(ref)},
+            me_ptr{&me}
+        {}
+        //!\}
+
+        /*!\name Iterators
+         * \brief This column type is not const-iterable.
+         * \{
+         */
+        //!\brief Returns an iterator to the begin of the column.
+        constexpr auto begin() noexcept
+        {
+            assert(me_ptr != nullptr);
+            return iterator_type{*this};
+        }
+
+        //!\brief Deleted begin for const-qualified alignment-columns.
+        constexpr auto begin() const noexcept = delete;
+
+        //!\brief Deleted begin for const-qualified alignment-columns.
+        constexpr auto cbegin() const noexcept = delete;
+
+        //!\brief Returns an iterator to the end of the column.
+        constexpr sentinel end() noexcept
+        {
+            return ref.end();
+        }
+
+        //!\brief Deleted end for const-qualified alignment-columns.
+        constexpr sentinel end() const noexcept = delete;
+
+        //!\brief Deleted end for const-qualified alignment-columns.
+        constexpr sentinel cend() const noexcept = delete;
+        //!\}
+
+    private:
+        //!\brief The aliased alignment-column.
+        view_type ref{};
+        //!\brief Pointer to the derived type.
+        derived_t * me_ptr{};
+    }; // class alignment_column_type
+
+    /*!\brief A column iterator over the alignment matrix.
+     *
+     * \details
+     *
+     * This iterator enables the iteration over the underlying alignment matrix in column-major-order.
+     * When dereferenced the iterator returns a seqan3::detail::alignment_matrix_column_major_range::column_type.
+     * The initialisation of the alignment-column depends on the derived type.
+     */
+    class iterator_type
```

Comment:
there are many reasons. The most significant one is that in the algorithm you need to know when you start the next column since you have to process the cells at the top of the column differently.